src/main/kotlin/no/nav/familie/ks/sak/internal/UtledFagsakSomHarUtbetalingSammeMånedSomBarnehagestart.kt

```diff
@@ -0,0 +1,77 @@
+package no.nav.familie.ks.sak.internal
+
+import no.nav.familie.ks.sak.common.util.toYearMonth
+import no.nav.familie.ks.sak.integrasjon.secureLogger
+import no.nav.familie.ks.sak.kjerne.behandling.BehandlingService
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårsvurderingRepository
+import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelseRepository
+import no.nav.familie.ks.sak.kjerne.fagsak.domene.FagsakRepository
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.PersonopplysningGrunnlagService
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.YearMonth
+
+@Service
+class UtledFagsakSomHarUtbetalingSammeMånedSomBarnehagestart(
+    private val fagsakRepository: FagsakRepository,
+    private val behandlingService: BehandlingService,
+    private val andelerTilkjentYtelseRepository: AndelTilkjentYtelseRepository,
+    private val vilkårsvurderingRepository: VilkårsvurderingRepository,
+    private val personopplysningGrunnlagService: PersonopplysningGrunnlagService,
+) {
+    fun utledFagsakerSomHarUtbetalingSammeMånedSomBarnehagestart() {
+        // Ved start av fulltidplass barnehage så vil andelen ende. Relevante andeler er andeler som sluttes å utbetales i august - des 2024
+        val relevanteSluttMånederForAndel = (8..12).map { YearMonth.of(2024, it) }
+
+        // Hent fagsaker som har sist iverksatt behandling som hadde en andel som varte fram til august - des 2024.
+        val fagsaker = fagsakRepository.finnFagsakerSomHarSistIverksattBehandlingMedUtbetalingSomStopperMellomAugOgDes2024()
+
+        // Henter sist iverksatte behandling igjen for fagsaker. Dette bare for å være 100% sikker, slik at vi ikke lener oss for mye på selve SQL spørringen.
+        val sistIverksatteBehandlinger = fagsaker.mapNotNull { behandlingService.hentSisteBehandlingSomErIverksatt(it.id) }
```

Comment:
Sparer også en del minne hvis man f.eks. mapper til .map { Pair(it.behandlingId, it.fagsak.id) }